gtsfm/two_view_estimator.py

```diff
@@ -357,6 +401,75 @@ def run_2view(
             post_isp_report,
         ) = self.processor.run_inlier_support(post_ba_i2Ri1, post_ba_i2Ui1, post_ba_v_corr_idxs, post_ba_report)
 
+        # Calculate computation time
+        import socket
+        import time
+        start_time = time.time()
+        worker_name = socket.gethostname()
+        
+        # Store results in the database
+        if self.db:
+            # Determine if the result is successful
+            success = (post_isp_i2Ri1 is not None and post_isp_i2Ui1 is not None)
+            
+            # Serialize rotation matrix and translation direction
+            rotation_matrix = None
+            translation_direction = None
+            if success:
+                # Serialize rotation matrix
+                if hasattr(post_isp_i2Ri1, 'matrix'):
+                    rotation_matrix = self.serialize_matrix(post_isp_i2Ri1.matrix())
+                else:
+                    rotation_matrix = self.serialize_matrix(post_isp_i2Ri1)
+                
+                # Serialize translation direction
+                if hasattr(post_isp_i2Ui1, 'point3'):
+                    translation_direction = self.serialize_matrix(post_isp_i2Ui1.point3())
+                else:
+                    translation_direction = self.serialize_matrix(post_isp_i2Ui1)
+            
+            # Get number of verified correspondences and inlier ratio
+            verified_corr_count = len(post_isp_v_corr_idxs) if post_isp_v_corr_idxs is not None else 0
+            inlier_ratio = post_isp_report.inlier_ratio_est_model if post_isp_report else 0.0
+            
+            # Insert result into the database
+            insert_query = """
+            INSERT INTO two_view_results 
+            (i1, i2, timestamp, verified_corr_count, inlier_ratio, rotation_matrix, 
+            translation_direction, success, computation_time, worker_name)
+            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
+            """
+            
+            self.db.execute(
+                insert_query,
+                (keypoints_i1.image_id, keypoints_i2.image_id, datetime.now(), verified_corr_count, inlier_ratio, 
+                 rotation_matrix, translation_direction, success, time.time() - start_time, worker_name)
+            )
+            
+            # save detailed report
```

Comment:
Maybe even out both queries in two additional methods, and top-method calls both. Clarity